### ISSUE TITLE
Choices do not blank field when data isn't part of choices

### ISSUE THREAD

**Issue by corradio** [bhch/react-json-form#72]
Consider the following schema from the playground:
```json
{
  "type": "object",
  "keys": {
    "country": {
      "type": "string",
      "choices": [
        "Australia",
        "India",
        "United Kingdom",
        "United States"
      ]
    }
  }
}
```

When the data `{"country": "XXX"}` is passed, I'd expect the rendered UI selector to be blank (i.e. same as if the data was empty) as the passed input is not part of the choices. Right now, it selects the first item on the list.

**Comment by corradio** [comment]
Thanks for the explanation, that makes sense. Then maybe a cleaner fix
would be to make sure the values after form submit reflect the first non
disabled option? Right now it returns the value of the element that wasn’t
in the list.

**Comment by bhch** [comment]
Right now, we don't implicitly modify the originally supplied data. So, unless the user manually selects a different option, the value in the data stays the same `XXX`.

But maybe we can make an exception in this case and modify the value to an empty string in case of no match.

Also, indirectly related to this, I've just noticed that currently there's no validation for choices on string type neither on frontend nor in the django app. So, I'll be implementing that as well.

**Comment by bhch** [comment]
Fix released in [v2.8.3](https://github.com/bhch/react-json-form/releases/tag/v2.8.3) and in [`django-jsonform` v2.17.3](https://github.com/bhch/django-jsonform/releases/tag/v2.17.3).

Please update and test it out.

Thanks for creating this issue.

**Comment by corradio** [comment]
Works like a charm - thanks for fixing!
